File: 5.Longest_Palindromic_Substring/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if s == s[::-1]:
            return s
        t, n = 0, 1
        for i in range(1, len(s)):
            l, r = i - n, i + 1
            s1 = s[l-1 : r]
            if l >= 1 and s1 == s1[::-1]:
                n += 2
                t = l - 1
            else:
                s2 = s[l:r]
                if s2 == s2[::-1]:
                    n += 1
                    t = l
        return s[t:t+n]

File: 5.Longest_Palindromic_Substring/solution.rs
impl Solution {
    pub fn longest_palindrome(s: String) -> String {
        let sb = s.as_bytes();
        let mut pos = (0, 0);

        for i in 0..sb.len() * 2 {
            let (mut l, mut r) = (i / 2, i / 2 + (i % 2 != 0) as usize);
            while l <= r && l < sb.len() && r < sb.len() {
                if sb[l] == sb[r] {
                    if r - l > pos.1 - pos.0 { pos = (l, r) }
                } else {
                    break
                }
                l -= 1;
                r += 1;
            }
        }
        s[pos.0..=pos.1].to_string()
    }
}
